js/src/file-input.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * --------------------------------------------------------------------------
+ * Bootstrap (v5.0.0-alpha1): file-input.js
+ * Licensed under MIT (https://github.com/twbs/bootstrap/blob/main/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+import { getjQuery } from './util/index'
+import Data from './dom/data'
+import EventHandler from './dom/event-handler'
+import SelectorEngine from './dom/selector-engine'
+
+/**
+ * ------------------------------------------------------------------------
+ * Constants
+ * ------------------------------------------------------------------------
+ */
+
+const NAME = 'fileInput'
+const VERSION = '5.0.0-alpha1'
+const DATA_KEY = 'bs.file-input'
+const EVENT_KEY = `.${DATA_KEY}`
+const DATA_API_KEY = '.data-api'
+
+const SELECTOR_DATA_TOGGLE = '[data-toggle="file-input"]'
+const SELECTOR_FILE_INPUT = '.form-file-input'
+const SELECTOR_FILE_INPUT_LABEL = '.form-file-text'
+const SELECTOR_FORM = 'form'
+
+const EVENT_CLICK_DATA_API = `click${EVENT_KEY}${DATA_API_KEY}`
+const EVENT_CHANGE_DATA_API = `change${EVENT_KEY}${DATA_API_KEY}`
+const EVENT_RESET_DATA_API = `reset${EVENT_KEY}${DATA_API_KEY}`
+
+const fileApi = Boolean(window.File)
```

Comment:
Shouldn't this be in capitals since it's a constant? Also, is the variable name right? It seems like we are feature detecting.  Just wondering :)

Reply:
yep it's a feature detection, I have to check that on Can I Use because nowadays File API should be supported everywhere

Reply:
![image](https://user-images.githubusercontent.com/1689750/85113101-0cfb2900-b217-11ea-9541-c0095fdb3543.png)
finally FileList API isn't supported by Opera Mini so we should keep the feature detection